backend/src/controllers/student.ts

```diff
@@ -0,0 +1,78 @@
+/* eslint-disable @typescript-eslint/no-misused-promises */
+/**
+ * Functions that process task route requests.
+ */
+
+import { RequestHandler } from "express";
+import { validationResult } from "express-validator";
+
+import { CustomError } from "../errors/errors";
+import { errorHandler } from "../errors/handler";
+import StudentModel from "../models/student";
+
+export const createStudent: RequestHandler = async (req, res, next) => {
+  type contact = {
+    lastName: string;
+    firstName: string;
+    email: string;
+    phoneNumber: string;
+  };
+
+  type typedModel = {
+    student: contact;
+    emergency: contact;
+    serviceCoordinator: contact;
+    location: string;
+    medication: string;
+    birthday: string;
+    intakeDate: string;
+    tourDate: string;
+    prog1: string;
+    prog2: string;
+    dietary: string;
+    otherString: string;
+  };
+
+  const errors = validationResult(req);
+
+  const {
+    student,
+    emergency,
+    serviceCoordinator,
+    location,
+    medication,
+    birthday,
+    intakeDate,
+    tourDate,
+    prog1,
+    prog2,
+    dietary,
+    otherString,
+  }: typedModel = req.body as typedModel; // ?????????
+
+  try {
+    let errornum = 0;
+    for (const error of errors.array()) {
+      errorHandler(new CustomError(errornum++, 400, error.msg as string), req, res);
```

Comment:
Let's not use the CustomErrors in this way. We need to create the errors in the errors folders. See [this](https://github.com/TritonSE/ALUM-Mobile-Application/tree/main/backend/src/errors) as an example. We want to initialize the error handler like [this](https://github.com/TritonSE/ALUM-Mobile-Application/blob/main/backend/src/app.ts#L39). Store the errors in ../errors/vallidation.ts. Once this is setup properly we only need to throw that custom error.